pg_search/src/aggregate/mod.rs

```diff
@@ -262,32 +375,77 @@ impl ParallelWorker for ParallelAggregationWorker<'_> {
             .object::<Config>(1)
             .expect("wrong type for config")
             .expect("missing config value");
-        let agg_req_bytes = state_manager
+        let base_query_bytes = state_manager
             .slice::<u8>(2)
-            .expect("wrong type for agg_req_bytes")
-            .expect("missing agg_req_bytes value");
-        let query_bytes = state_manager
+            .expect("wrong type for base_query_bytes")
+            .expect("missing base_query_bytes value");
+        let aggregate_types_bytes = state_manager
             .slice::<u8>(3)
-            .expect("wrong type for query_bytes")
-            .expect("missing query_bytes value");
+            .expect("wrong type for aggregate_types_bytes")
+            .expect("missing aggregate_types_bytes value");
+        let grouping_columns_bytes = state_manager
+            .slice::<u8>(4)
+            .expect("wrong type for grouping_columns_bytes")
+            .expect("missing grouping_columns_bytes value");
+        let orderby_info_bytes = state_manager
+            .slice::<u8>(5)
+            .expect("wrong type for orderby_info_bytes")
+            .expect("missing orderby_info_bytes value");
+        let limit_offset_bytes = state_manager
+            .slice::<u8>(6)
+            .expect("wrong type for limit_offset_bytes")
+            .expect("missing limit_offset_bytes value");
+        let agg_json_bytes = state_manager
+            .slice::<u8>(7)
+            .expect("wrong type for agg_json_bytes")
+            .expect("missing agg_json_bytes value");
         let segment_ids = state_manager
-            .slice::<(SegmentId, NumDeletedDocs)>(4)
+            .slice::<(SegmentId, NumDeletedDocs)>(8)
             .expect("wrong type for segment_ids")
             .expect("missing segment_ids value");
         let ambulkdelete_epoch = state_manager
-            .object::<u32>(5)
+            .object::<u32>(9)
             .expect("wrong type for ambulkdelete_epoch")
             .expect("missing ambulkdelete_epoch value");
 
-        let aggregation = serde_json::from_slice::<Aggregations>(agg_req_bytes)
-            .expect("agg_req_bytes should deserialize into an Aggregations");
-        let query = serde_json::from_slice::<SearchQueryInput>(query_bytes)
-            .expect("query_bytes should deserialize into an SearchQueryInput");
+        let base_query = serde_json::from_slice::<SearchQueryInput>(base_query_bytes)
+            .expect("base_query_bytes should deserialize into SearchQueryInput");
+
+        // Check if this is a JSON aggregation or filter aggregation
```

Comment:
Ditto the comment above about these being separate fields: if this were an `enum`, you'd directly `match` here.